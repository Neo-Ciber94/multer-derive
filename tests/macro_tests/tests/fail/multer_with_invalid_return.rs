use multer_derive::{Error, FromMultipart, MultipartForm, FormContext};

#[derive(FromMultipart)]
struct MyStruct {
    #[multer(with = "text_from_multipart")]
    text: Text,
    number: u32,
}

struct Text(String);

fn text_from_multipart(_multipart: &MultipartForm, _ctx: FormContext<'_>) -> Result<Text, ()> {
    todo!()
}

fn main() {}
