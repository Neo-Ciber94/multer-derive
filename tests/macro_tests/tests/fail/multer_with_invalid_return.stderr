warning: unused import: `Error`
 --> tests/fail/multer_with_invalid_return.rs:1:21
  |
1 | use multer_derive::{Error, FromMultipart, MultipartForm, FormContext};
  |                     ^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

error[E0277]: `?` couldn't convert the error to `multer_derive::Error`
 --> tests/fail/multer_with_invalid_return.rs:3:22
  |
3 | #[derive(FromMultipart)]
  |                      ^ the trait `From<()>` is not implemented for `multer_derive::Error`
  |
  = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
  = help: the trait `From<multer_derive::multer::Error>` is implemented for `multer_derive::Error`
  = note: required for `Result<MyStruct, multer_derive::Error>` to implement `FromResidual<Result<Infallible, ()>>`
  = note: this error originates in the derive macro `FromMultipart` (in Nightly builds, run with -Z macro-backtrace for more info)
